lib/core/transpiler_session/header_info.cpp

```diff
@@ -27,19 +35,67 @@ void InclusionDirectiveCallback::InclusionDirective(clang::SourceLocation hashLo
         return;
     }
 
-    deps.topLevelDeps.push_back(HeaderDep{
-        .hashLoc = hashLoc,
-        .includeTok = includeTok,
-        .fileName = fileName.str(),
-        .isAngled = isAngled,
-        .filenameRange = filenameRange,
-        .file = file,
-        .searchPath = searchPath.str(),
-        .relativePath = relativePath.str(),
-        .imported = imported,
-        .fileType = fileType,
-
-    });
+    auto fileNameStr = fileName.str();
+    auto isIntrinsic = overrideExternalIntrinsic(_session, fileNameStr, file, sm);
+    if (isIntrinsic) {
+        _extIntrinsicFID = sm.getFileID(hashLoc);
+    }
+
+    // INFO: force instrinsic includes to be system includes
+    //       they will be removed & attached to final transpiled source
+    if (_isInExternalIntrinsic) {
```

Comment:
Those two branches differ only in fileType.
Let have local function to define correct one and use it during pushing back.

Reply:
Nope It pushs to the different pools also